morango/models.py

```diff
@@ -222,11 +229,40 @@ class Meta:
 class DatabaseMaxCounter(AbstractCounter):
     """
     ``DatabaseMaxCounter`` is used to keep track of what data this database already has across all
-    instances for a particular filter. Whenever 2 morango instances sync with each other we keep track
-    of those filters, as well as the maximum counter we received for each instance during the sync session.
+    instances for a particular partition prefix. Whenever 2 morango instances sync with each other we keep track
+    of those partition prefixes from the filters, as well as the maximum counter we received for each instance during the sync session.
     """
 
-    filter = models.TextField()
+    partition = models.CharField(max_length=128, default="")
+
+    @classmethod
+    def calculate_filter_max_counters(cls, filters):
```

Comment:
After merging https://github.com/learningequality/morango/pull/21, this can be updated to take an instance of `Filter`, and then we won't need to split on "\n".